packages/tooltip/src/Tooltip.tsx

```diff
@@ -11,26 +11,57 @@ import { useMergeRefs } from '@lidofinance/hooks'
 import { isElement } from 'react-is'
 import { TooltipProps } from './types'
 
+const KEEP_TIMEOUT_DEFAULT = 150
+
 function Tooltip(props: TooltipProps, ref?: ForwardedRef<HTMLDivElement>) {
-  const { children, title, ...rest } = props
+  const {
+    title,
+    keepWhileHovering,
+    keepTimeout = KEEP_TIMEOUT_DEFAULT,
+    children,
+    ...rest
+  } = props
+
   const [state, setState] = useState(false)
+  const keepTimeoutRef = useRef(-1)
 
   const child = Children.only(children)
   if (!isElement(child)) throw new Error('Child must be a React element')
 
   const anchorRef = useRef<HTMLElement>(null)
   const mergedRef = useMergeRefs([child.ref, anchorRef])
 
+  const handleMouseEnterKeeping = () => {
+    if (keepTimeoutRef.current !== -1) {
+      clearTimeout(keepTimeoutRef.current)
+      keepTimeoutRef.current = -1
+    }
+  }
+
+  const handleMouseEnterLeaveKeeping = () => {
+    keepTimeoutRef.current = setTimeout(() => {
+      setState(false)
+      keepTimeoutRef.current !== -1
+    }, keepTimeout)
+  }
+
   return (
     <>
       {cloneElement(child, {
         ref: mergedRef,
         onMouseEnter(event: React.MouseEvent<HTMLElement, MouseEvent>) {
+          if (keepWhileHovering) {
+            handleMouseEnterKeeping()
+          }
           setState(true)
           child.props.onMouseEnter?.(event)
         },
         onMouseLeave(event: React.MouseEvent<HTMLElement, MouseEvent>) {
-          setState(false)
+          if (keepWhileHovering) {
+            handleMouseEnterLeaveKeeping()
```

Comment:
I don't quite understand the logic. When you move the mouse away from an element, instead of hiding popup, the popup will display another 150 milliseconds?

Reply:
Yep, it's not really noticable for user, but we need this fast gap to prevent disappearing while cursor is moving between root trigger and popover body elements.